types/react-interactive/index.d.ts

```diff
@@ -0,0 +1,85 @@
+// Type definitions for react-interactive 0.9
+// Project: https://github.com/rafrex/react-interactive
+// Definitions by: Daniel <https://github.com/DanudeSandstorm>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+import { ReactNode, ReactElement, Component, SyntheticEvent, CSSProperties } from 'react';
+
+export {};
+
+export type iState = 'normal' | 'hover' | 'hoverActive' | 'touchActive' | 'keyActive';
+export type focus = false | 'tab' | 'mouse' | 'touch';
+export type clickType = "mouseClick" | "tapClick" | "keyClick";
+
+export type State = {focus: focus, iState: iState};
+
+type Active = {
+    active?: CSSProperties;
+} | {
+    hoverActive?: CSSProperties;
+    touchActive?: CSSProperties;
+    keyActive?: CSSProperties;
+};
+
+type Focus = {
+    focus?: CSSProperties;
+} | {
+    focusFromTab?: CSSProperties;
+    focusFromMouse?: CSSProperties;
+    focusFromTouch?: CSSProperties;
+};
+
+type InteractiveProps = Focus & Active & {
```

Comment:
Seems like it would be nice to export this type, in case someone wants to write a component that composes this one